components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java

```diff
@@ -0,0 +1,114 @@
+import org.wso2.carbon.identity.application.authentication.handler.session.ActiveSessionsLimitHandler;
+import org.wso2.carbon.user.core.service.RealmService;
+
+/**
+ * OSGi declarative services component which handles registration and de-registration of Session Handler.
+ */
+@Component(
+        name = "identity.application.handler.session.component",
+        immediate = true
+)
+public class ActiveSessionsLimitHandlerServiceComponent {
+
+    private static final Log log = LogFactory.getLog(ActiveSessionsLimitHandlerServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext ctxt) {
+
+        try {
+            ActiveSessionsLimitHandler activeSessionsLimitHandler = new ActiveSessionsLimitHandler();
+            ctxt.getBundleContext().registerService(ApplicationAuthenticator.class.getName(), activeSessionsLimitHandler, null);
+            if (log.isDebugEnabled()) {
+                log.debug("ActiveSessionsLimitHandler bundle is activated");
+            }
+        } catch (Throwable e) {
+            log.error("ActiveSessionsLimitHandler Authenticator bundle activation Failed", e);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext ctxt) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("ActiveSessionsLimitHandler bundle is deactivated");
+        }
+    }
+
+    @Reference(
+            name = "realm.service",
+            service = RealmService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetRealmService"
+    )
+    protected void setRealmService(RealmService realmService) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Setting the Realm Service");
+        }
+        ActiveSessionsLimitHandlerServiceHolder.getInstance().setRealmService(realmService);
+    }
+
+    protected void unsetRealmService(RealmService realmService) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("UnSetting the Realm Service");
+        }
+        ActiveSessionsLimitHandlerServiceHolder.getInstance().setRealmService(null);
+    }
+
+    @Reference(
+            name = "org.wso2.carbon.identity.application.authentication.framework.UserSessionManagementService",
+            service = UserSessionManagementService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetUserSessionManagementService"
+    )
+    protected void setUserSessionManagementService(UserSessionManagementService userSessionManagementService) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("UserSessionManagementService is set in the conditional authentication user functions bundle");
+        }
+        ActiveSessionsLimitHandlerServiceHolder.getInstance().setUserSessionManagementService(userSessionManagementService);
```

Comment:
Should be exceeding 120 lines count.

Reply:
Resolved in https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/68/commits/aa9404e4bdae50bedd69f6a1492773da1a76bc80